### DISCUSSION TITLE
Unable to connect

### DISCUSSION THREAD

**Discussion by ekr3peeK** [socketio/socket.io#3980] [Q&A] [answered]
Even with the most basic configuration, where I am not setting any other options, I am unable to connect with the client to the server.

I enabled logging on the client side, and this is the output I am getting:

```
engine.io-client:polling polling +0ms
engine.io-client:polling-xhr xhr poll +0ms 
engine.io-client:polling-xhr xhr open GET: https://myurl.com:3200/socket.io/?EIO=4&transport=polling&t=NeVWs2q +1ms
engine.io-client:polling-xhr xhr data null +1ms
engine.io-client:socket setting transport polling +5ms 
socket.io-client:manager connect attempt will timeout after 20000 +8ms 
socket.io-client:manager readyState opening +1ms
socket.io-client:manager readyState opening +61ms 
engine.io-client:polling polling got data 96:0{"sid":"pT-a7VjT8qhKVTBQAAD8","upgrades":["websocket"],"pingInterval":25000,"pingTimeout":5000}2:40 +144ms 
engine.io-client:socket socket receive: type "error", data "parser error" +141ms
```

I've followed the troubleshooting guide for the connection, step-by-step, and I am out of ideas on how to proceed, or what could be the error. Based on the log in my browser, it seems to me that the client is unable to parse the data received by the server. Semantically, the data received, `96:0{"sid":"pT-a7VjT8qhKVTBQAAD8","upgrades":["websocket"],"pingInterval":25000,"pingTimeout":5000}2:40` looks to be ok to me (at least this is simmilar to the data I've received while using v2 version). Anyone has any idea what is causing this error, or a way I could try to fix this?

**Comment by darrachequesne** [comment, marked as answer]
Hi! It looks like a case of https://socket.io/docs/v4/troubleshooting-connection-issues/#The-client-is-not-compatible-with-the-version-of-the-server

Could you please check the version of your server/client?

**Reply by ekr3peeK** [reply]
I've managed to figure this out in the end, it was an error on my part. My router was forwarding the port to the old dev server, where the v2 version of socket.io server was still running, and that's why I couldn't connect. Silly mistake:( Thanks for taking the time to try and help though!
